runtime-modules/proposals/codex/src/lib.rs

```diff
@@ -58,40 +58,92 @@ mod types;
+};
 
-// 'Set working group budget capacity' proposal limit
-const WORKING_GROUP_BUDGET_CAPACITY_MAX_VALUE: u32 = 5_000_000;
-// Max allowed value for 'spending' proposal
+use common::working_group::WorkingGroup;
+
+// Max allowed value for 'Funding Request' proposal
 const MAX_SPENDING_PROPOSAL_VALUE: u32 = 5_000_000_u32;
-// Max validator count for the 'set validator count' proposal
+// Max validator count for the 'Set Max Validator Count' proposal
 const MAX_VALIDATOR_COUNT: u32 = 100;
 
+/// Proposal codex WeightInfo.
+/// Note: This was auto generated through the benchmark CLI using the `--weight-trait` flag
+pub trait WeightInfo {
+    fn execute_signal_proposal(i: u32) -> Weight;
+    fn create_proposal_signal(i: u32, t: u32) -> Weight;
+    fn create_proposal_runtime_upgrade(i: u32) -> Weight;
+    fn create_proposal_funding_request(t: u32, d: u32) -> Weight;
+    fn create_proposal_set_max_validator_count() -> Weight;
+    fn create_proposal_create_working_group_lead_opening(i: u32, d: u32) -> Weight;
+    fn create_proposal_fill_working_group_lead_opening(t: u32) -> Weight;
+    fn create_proposal_update_working_group_budget(d: u32) -> Weight;
+    fn create_proposal_decrease_working_group_lead_stake(t: u32, d: u32) -> Weight;
+    fn create_proposal_slash_working_group_lead() -> Weight;
+    fn create_proposal_set_working_group_lead_reward(t: u32) -> Weight;
+    fn create_proposal_terminate_working_group_lead(t: u32) -> Weight;
+    fn create_proposal_amend_constitution(i: u32, t: u32) -> Weight;
+    fn create_proposal_cancel_working_group_lead_opening() -> Weight;
+    fn create_proposal_set_membership_price(d: u32) -> Weight;
+    fn create_proposal_set_council_budget_increment(t: u32, d: u32) -> Weight;
+    fn create_proposal_set_councilor_reward(d: u32) -> Weight;
+    fn create_proposal_set_initial_invitation_balance(t: u32) -> Weight;
+    fn create_proposal_set_initial_invitation_count() -> Weight;
+    fn create_proposal_set_membership_lead_invitation_quota() -> Weight;
+    fn create_proposal_set_referral_cut(t: u32, d: u32) -> Weight;
+    fn update_working_group_budget_positive_forum() -> Weight;
+    fn update_working_group_budget_negative_forum() -> Weight;
+    fn update_working_group_budget_positive_storage() -> Weight;
+    fn update_working_group_budget_negative_storage() -> Weight;
+    fn update_working_group_budget_positive_content() -> Weight;
+    fn update_working_group_budget_negative_content() -> Weight;
+    fn update_working_group_budget_positive_membership() -> Weight;
+    fn update_working_group_budget_negative_membership() -> Weight;
+    fn funding_request() -> Weight;
+}
+
+type WeightInfoCodex<T> = <T as Trait>::WeightInfo;
+
 /// 'Proposals codex' substrate module Trait
 pub trait Trait:
     frame_system::Trait
     + proposals_engine::Trait
     + proposals_discussion::Trait
     + common::Trait
+    + council::Trait
     + staking::Trait
+    + working_group::Trait<ForumWorkingGroupInstance>
```

Comment:
I could fix this by adding a hook to the runtime that allows us to query and set the working group budget. Also, this would fix having the working group instances definitions in the working group since this way I would only need them in the runtime and I can move them back there. Do you think this is an acceptable way of handling this?

Reply:
Yes, I also thought of the runtime hook. Let's create a prototype to see how it goes.